lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,48 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  This task runs predefined upgrade steps.
+
+  Examples:
+    rake upgrade:run
+    rake upgrade:mark_as_ran TASK_NAME=some_rake_task,another_rake_task
+
+END_DESC
+
+namespace :upgrade do
+  task :run => :environment do
+    ENV['FOREMAN_UPGRADE'] = '1'
```

Comment:
This allows rake tasks to know if they are running as part of foreman-maintain.   A lot of the katello rake tasks do not make any changes by default (as they may be run manually), but allow a COMMIT=true option to actually commit the changes.  Instead of allowing specific options to be passed to specific rake tasks we, we ended up setting a specific ENV variable here so tasks can know that they are being run within this upgrade process and not by a user.  There was discussion around this above.